File: src/main/java/xyz/terrific/bot/command/Command.java
package xyz.terrific.bot.command;

import xyz.terrific.bot.Bot;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public abstract class Command {

    public String                   name;
    public String                   description;
    public String                   syntax;
    public List<String>             aliases;



    public Command(String name, String description) {
        super();
        this.name = name;
        this.description = description;
        this.syntax = "";
        this.aliases = Collections.singletonList(name);
    }

    public Command(String name, String description, String[] aliases) {
        super();
        this.name = name;
        this.description = description;
        this.syntax = "";
        this.aliases = Arrays.asList(aliases);
    }

    public Command(String name, String description, String syntax, String[] aliases) {
        super();
        this.name = name;
        this.description = description;
        this.syntax = syntax;
        this.aliases = Arrays.asList(aliases);
    }



    public abstract void onCommand(String[] args, String command, Bot bot);
    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }
    public String getSyntax() {
        return syntax;
    }
    public void setSyntax(String syntax) {
        this.syntax = syntax;
    }
    public List<String> getAliases() {
        return aliases;
    }
    public void setAliases(List<String> aliases) {
        this.aliases = aliases;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/java/xyz/terrific/bot/command/Command.kt
package xyz.terrific.bot.command

import xyz.terrific.bot.Bot
import java.util.*

/**
 * @author TerrificTable55
 * @version 1.0
 */
abstract class Command {
    var name: String
    var description: String
    var syntax: String
    var aliases: List<String>

    /**
     * @use One of the three ways to define a new Command, takes a new name and description
     * @param name
     * @param description
     */
    constructor(name: String, description: String) : super() {
        this.name = name
        this.description = description
        syntax = ""
        aliases = listOf(name)
    }

    /**
     * @use Second way to define a new Command, takes a new name, description and aliases
     * @param name
     * @param description
     * @param aliases
     */
    constructor(name: String, description: String, aliases: Array<String?>) : super() {
        this.name = name
        this.description = description
        syntax = ""
        this.aliases = Arrays.asList(*aliases) as List<String>
    }

    /**
     * @use Third and last way to define a new Command, takes a new name, description, syntax and aliases
     * @param name
     * @param description
     * @param syntax
     * @param aliases
     */
    constructor(name: String, description: String, syntax: String, aliases: Array<String?>) : super() {
        this.name = name
        this.description = description
        this.syntax = syntax
        this.aliases = Arrays.asList(*aliases) as List<String>
    }

    /**
     * @use the `onCommand` function, its called when a command is sent to the bot, takes args, command and bot as arguments
     * @param args
     * @param command
     * @param bot
     */
    abstract fun onCommand(args: Array<String?>?, command: String?, bot: Bot?)
}
